src/puppetlabs/puppetdb/admin.clj

```diff
@@ -12,12 +14,41 @@
             [puppetlabs.puppetdb.middleware :as mid]
             [bidi.schema :as bidi-schema]
             [puppetlabs.puppetdb.schema :as pls]
-            [schema.core :as s]))
+            [schema.core :as s]
+            [slingshot.slingshot :refer [try+]]))
+
+(def clean-command-schema
+  {:version (s/eq 1)
+   :payload [(apply s/enum (map name cli-svc/clean-options))]})
+
+(def ^:private validate-clean-command (s/validator clean-command-schema))
+
+(defn- handle-clean-req [req clean]
+  (let [body (slurp (:body req))
+        command (json/parse-string body true)
+        invalid-info (try+
+                      (validate-clean-command command)
```

Comment:
I suppose - were you suggesting something like this?
```clojure
(def ^:private validate-clean-command (s/validator clean-command-schema))

(defn- invalid-clean-command-info [command]
  (try+
   (validate-clean-command command)
   nil
   (catch [:type :schema.core/error] ex
     ex)))

(defn- handle-clean-req [req clean]
  (let [body (slurp (:body req))
        command (json/parse-string body true)
        invalid-info (invalid-clean-command-info command)]
    (cond
      invalid-info
      (http/bad-request-response (tru "Invalid command {0}: {1}"
                                      (pr-str body)
                                      (pr-str (:error invalid-info))))
...
```
If so, then sure, if you prefer that.